boxoffice/views/order.py

```diff
@@ -0,0 +1,198 @@
+from boxoffice.models import Order, OnlinePayment, PaymentTransaction, User, PaymentFailError
+from boxoffice.extapi import razorpay
+from forms import LineItemForm, BuyerForm
+from boxoffice.mailclient import send_invoice_email
+from utils import xhr_only, api_result
+
+redis_connection = Redis()
+boxofficeq = Queue('boxoffice', connection=redis_connection)
+
+
+def jsonify_line_items(line_items):
+    """
+    Serializes and return line items in the format:
+    {item_id: {'quantity': Y, 'final_amount': Z, 'discounted_amount': Z, 'discount_policy_ids': ['d1', 'd2']}}
+    """
+    items_json = dict()
+    for line_item in line_items:
+        if not items_json.get(unicode(line_item.item_id)):
+            items_json[unicode(line_item.item_id)] = {'quantity': 0, 'final_amount': decimal.Decimal(0), 'discounted_amount': decimal.Decimal(0), 'discount_policy_ids': []}
+        if not items_json[unicode(line_item.item_id)].get('final_amount'):
+            items_json[unicode(line_item.item_id)]['final_amount'] = decimal.Decimal(0)
+        items_json[unicode(line_item.item_id)]['final_amount'] += line_item.base_amount - line_item.discounted_amount
+        items_json[unicode(line_item.item_id)]['discounted_amount'] += line_item.discounted_amount
+        items_json[unicode(line_item.item_id)]['quantity'] += 1
+        if line_item.discount_policy_id and line_item.discount_policy_id not in items_json[unicode(line_item.item_id)]['discount_policy_ids']:
+            items_json[unicode(line_item.item_id)]['discount_policy_ids'].append(line_item.discount_policy_id)
+    return items_json
+
+
+@app.route('/kharcha', methods=['OPTIONS', 'POST'])
+@xhr_only
+@cross_origin(origins=ALLOWED_ORIGINS)
+def kharcha():
+    """
+    Accepts JSON containing an array of line_items, with the quantity and item_id set for each line_item.
+
+    Returns JSON of line items in the format:
+    {item_id: {'quantity': Y, 'final_amount': Z, 'discounted_amount': Z, 'discount_policy_ids': ['d1', 'd2']}}
+    """
+    line_item_forms = LineItemForm.process_list(request.json.get('line_items'))
+    discount_coupons = request.json.get('discount_coupons')
+    if not line_item_forms:
+        return api_result(400, 'Invalid items')
+    # Make line item splits and compute amounts and discounts
+    line_items = LineItem.calculate([{'item_id': li_form.data.get('item_id')}
+        for li_form in line_item_forms
+        for _ in range(li_form.data.get('quantity'))], coupons=discount_coupons)
+    items_json = jsonify_line_items(line_items)
+    order_final_amount = sum([values['final_amount'] for values in items_json.values()])
+    return jsonify(line_items=items_json, order={'final_amount': order_final_amount})
+
+
+@app.route('/<organization>/<item_collection>/order',
+           methods=['GET', 'OPTIONS', 'POST'])
+@load_models(
+    (Organization, {'name': 'organization'}, 'organization'),
+    (ItemCollection, {'name': 'item_collection'}, 'item_collection')
```

Comment:
If `ItemCollection` is loaded regardless of what organization it is under, why do you need to load the organization?